src/gui/guiTable.cpp

```diff
@@ -373,8 +376,18 @@ void GUITable::setTable(const TableOptions &options,
 				if (row->content_index >= 0)
 					image = m_images[row->content_index];
 
+				row->image_scale = 1.0f;
+				row->content_width = 0;
+				if (image) {
+					f32 max_image_scale = (f32)m_rowheight / (f32)image->getOriginalSize().Height;
+					// Scale with display density and make sure it fits into the row
+					row->image_scale = std::min(density, max_image_scale);
+					// When upscaling, only allow integer multiples
```

Comment:
This describes what the code does, but not why it needs to do it.

```suggestion
					// When upscaling, fractional factors do not look good, thus limit to integers.
```